xbmc/filesystem/CurlFile.cpp

```diff
@@ -535,6 +535,11 @@ void CCurlFile::SetCommonOptions(CReadState* state)
     state->m_curlAliasList = g_curlInterface.slist_append(state->m_curlAliasList, "ICY 200 OK");
   g_curlInterface.easy_setopt(h, CURLOPT_HTTP200ALIASES, state->m_curlAliasList);
 
+#if defined(TARGET_ANDROID) || defined(TARGET_DARWIN) || defined(TARGET_WINDOWS)
```

Comment:
1. I didn't find in curl docs which env var should I use.
2. env var is a paint for uwp. at least I will have to patch curl for new env var.

Reply:
osx works without the fix

Reply:
android?

Reply:
don't know for android
wanted to test ios, but nightlies are broken since #13722

Reply:
@afedchin I did not mean an env var directly for curl. I mean something like

std::string cacert = getenv(KODI_CERT);
if (cacert)
  g_curlInterface.easy_setopt(h, CURLOPT_CAINFO, cacert.c_str());

platform startup code can do setenv if env var not already set by user.

This way users can override the default like they can do on Linux

Reply:
We could update curl on Windows and set `CURL_CA_FALLBACK`. Then it should theoretically use the OpenSSL environment variable.

https://github.com/curl/curl/blob/master/CMakeLists.txt#L634

Reply:
@pkerling OpenSSL (which is used on windows) has no internal cert checks see https://curl.haxx.se/docs/ssl-compared.html

Reply:
Yes, which is why you do https://github.com/xbmc/xbmc/blob/af745eecbac8ee9d8e565e30f8c459f135f20e41/xbmc/platform/overrides/windowsstore/PlatformWin10.cpp#L37

I'm saying that curl should also use `SSL_CERT_FILE` when `CURL_CA_FALLBACK` is set.

Reply:
you mixed two different things. OSSL uses this CA (on uwp) to make possible to do TLS connectivity. Curl uses it for validation which OSSL can't do.
I'm not a supporter of unnecessary patches for dependencies. I'd prefer do not patch a lib if we can do things without it.

Reply:
seems like I got your opinion. But setting `SSL_CERT_FILE` doesn't work for uwp. We have to patch curl for uwp to make it works. We already do the same for OSSL because there's no other way.

Reply:
I don't follow.

> OSSL uses this CA (on uwp) to make possible to do TLS connectivity.

Not exactly. It allows making HTTPS requests from Python because Python also uses OpenSSL (but not curl) for HTTP connections and needs some certificates to validate the peer. Otherwise all connections would fail (like they do now for add-on updates). It has nothing to do with core (C++) Kodi at the moment.

> Curl uses it for validation which OSSL can't do.

Of course OpenSSL can validate certificates. It's just a matter of curl giving OpenSSL a CA store either explicitly (via `CURLOPT_CAINFO`) or allowing the library to use its default (with `CURL_CA_FALLBACK`), which for OpenSSL would imply using `SSL_CERT_FILE`.

> I'd prefer do not patch a lib if we can do things without it.

Activating `CURL_CA_FALLBACK` does not need a patch, you just enable the CMake option when building.

Reply:
> But setting SSL_CERT_FILE doesn't work for uwp. We have to patch curl for uwp to make it works.

You already patch OpenSSL to pick that up, right? So what would you have to patch in curl?

Reply:
> Activating CURL_CA_FALLBACK does not need a patch, you just enable the CMake option when building.

It requires a patch for curl-uwp here https://github.com/curl/curl/blob/master/lib/getenv.c#L33. There is no environment vars on uwp platform.

Reply:
curl does not read `SSL_CERT_FILE`, OpenSSL does.